### PR TITLE
[NEXUS-8177][CMA] Repository availability (online/offline)

### PR DESCRIPTION
**Pull request by adreghiciu**
https://issues.sonatype.org/browse/NEXUS-8177

### CODE REVIEW

**Comment by mrprescott** [comment]
+1. I agree it's faintly weird that the property is on viewFacet rather than on respository; I assume that's because repositories don't have events to pick up their new configuration?

**Comment by adreghiciu** [comment]
We discussed with Jason that the property eventually will have to go on repository itself yet we can do that later on

File: plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryCMAComponent.groovy [hunk 1]
```diff
@@ -103,6 +104,7 @@ extends DirectComponentSupport
         name: input.name,
         type: input.type,
         format: input.format,
+        online: input.facet(ViewFacet).online,
```
Comment (kellyrob99) [inline plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryCMAComponent.groovy]: are we guaranteed to have a ViewFacet here? If there's doubt, might want to use ?. as a guard

**Comment by kellyrob99** [comment]
+1

File: plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryCMAComponent.groovy (hunk 1 above)
Comment (adreghiciu) [inline plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/RepositoryCMAComponent.groovy]: ViewFacet is always available. facet() call will throw an exception if facet is not available

**Comment by adreghiciu** [comment]
Manually merged into CMA3 https://github.com/sonatype/nexus-oss/commit/ecdca3d6fab6e8b1710231bb73aa1acbfe2095dc

**Pull request closed without merging**